atmos_model.F90

```diff
@@ -262,13 +262,17 @@ subroutine update_atmos_radiation_physics (Atmos)
       call CCPP_step (step="timestep_init", nblks=Atm_block%nblks, ierr=ierr)
       if (ierr/=0)  call mpp_error(FATAL, 'Call to CCPP timestep_init step failed')
 
+      if (GFS_Control%do_sppt .or. GFS_Control%do_shum .or. GFS_Control%do_skeb .or. &
+          GFS_Control%lndp_type > 0  .or. GFS_Control%do_ca ) then
 !--- call stochastic physics pattern generation / cellular automata
-      call stochastic_physics_wrapper(GFS_control, GFS_data, Atm_block, ierr)
-      if (ierr/=0)  call mpp_error(FATAL, 'Call to stochastic_physics_wrapper failed')
+        call stochastic_physics_wrapper(GFS_control, GFS_data, Atm_block, ierr)
+        if (ierr/=0)  call mpp_error(FATAL, 'Call to stochastic_physics_wrapper failed')
+      endif
 
 !--- if coupled, assign coupled fields
 
-      if (.not. GFS_control%cplchm) then
+!     if (.not. GFS_control%cplchm) then
+      if (GFS_control%cplflx .or. GFS_control%cplwav2atm) then
         call assign_importdata(rc)
       endif
```

Comment:
This subroutine (`assign_importdata`) must be called every time, except when `cplchem` is true, not only when `cplflx` or `cplwav2atm` are true. For example when fv3 is coupled with JEDI.

Reply:
OK, I will revert; however, for standalone FV3, calling this routine is a waste.